src/controller/CHIPDeviceControllerFactory.cpp

```diff
@@ -150,6 +158,36 @@ CHIP_ERROR DeviceControllerFactory::InitSystemState(FactoryInitParams params)
     ReturnErrorOnFailure(Dnssd::Resolver::Instance().Init(stateParams.udpEndPointManager));
 #endif // CHIP_DEVICE_CONFIG_ENABLE_DNSSD
 
+    if (params.enableServerInteractions)
+    {
+        stateParams.caseServer = chip::Platform::New<CASEServer>();
+
+        ReturnErrorOnFailure(stateParams.caseServer->ListenForSessionEstablishment(
+            stateParams.exchangeMgr, stateParams.transportMgr, DeviceLayer::ConnectivityMgr().GetBleLayer(), stateParams.sessionMgr,
+            stateParams.fabricTable, stateParams.sessionIDAllocator));
+
+        //
+        // We need to advertise the port that we're listening to for unsolicited messages over UDP. However, we have both a IPv6
+        // and IPv6 endpoint to pick from. Since IPv6 is POR for Matter, let's go ahead and pick that port.
```

Comment:
Both IPv6 and IPv6?

In any case, please file an issue to see what we can do better here, because it's pretty clear that IPv6 is not going to work in many cases, especially when LLAs are involved....  Ideally we would bind the same port for both transports....

Reply:
> IPv6 is not going to work in many cases

That isn't an option from a spec standpoint right? IPv6 _has_ to work.

Reply:
If IPv6 doesn't work with LLA, it's always because of bad network configuration using VLAN forwarding, which is all proprietary stuff that has problems only when ppl go out of their way to be fancy. Matter does not claim to work on IPv4 at all.

Reply:
@mrjerryjohns is right, I don't see a reasonable path where IPv6 is somehow optional; there is just no way to land this with the spec we have today.

I also have no idea of how the LLA could be broken unless it had been explicitly broken with VLANs.   Maybe there is a chance that something else is getting broken by a non-compliant network that would squash the ethernet broadcasts, but IMO, in such cases we're better off failing early and signaling that to the customer, rather than failing very late (e.g. when a customer installs a different IPv6 subnet) and the debugging process is nearly impossible

Reply:
> Both IPv6 and IPv6?
> 
> In any case, please file an issue to see what we can do better here, because it's pretty clear that IPv6 is not going to work in many cases, especially when LLAs are involved.... Ideally we would bind the same port for both transports....

Can you talk about specific cases where "IPv6 is not going to work?"

Reply:
We have multiple user reports now from just the set of people working on this project of LLAs not working well with various WiFi routers. There is no VLAN anything involved in any of the cases I am aware of.  For some of the routers, rebooting the router helps ... for a while.  Then it stops working again.  This is very much not a "failing early" situation.

Yes, those are clearly buggy routers in some way.   No, we don't control what people have in their homes.

This is not how I want the world to be, obviously, and it's not the world the spec envisions, but based on what I see so far just in this small sample, I would expect that reliability of ipv6-only with LLAs will be not great.  Maybe this is a skewed sample, of course.

Yes, the spec assumes that ipv6 will always work.  That's not what we are seeing in practice...

Anyway, in case this was not clear this part is not blocking this PR.